src/utilities/modules/vaccines.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+/**
+ * Utility and helper methods for the vaccine
+ * module.
+ */
+
+/**
+ * Extracts breaches from a set of sensor logs.
+ * Breach: Sequential sensorLog objects whose temperature is outside
+ * the lower or uppound limits, form a breach. Each array returned
+ * contains all sensorLogs which form a breach INCLUDING the sensorLog
+ * prior to the breach, and after the breach, if they exist. There may
+ * be duplicate non-breach sensorLogs as breach delimiters if there is
+ * only one sensorLog between breaches.
+ *
+ * Method sorts the sensorLogs by timestamp, iterates and pushes onto a
+ * stack. Only one unBreached sensorLog is kept at the tail of the stack.
+ * All breached sensorLogs are pushed, until a non breached log, where this
+ * marks a breach has been found. Store these sensorLogs and reset the stack.
+ *
+ * @param  {Realm.results} sensorLogs array of sensorLog objects
+ * @return {Array<Realm.results<SensorLog>} A 2D array of Realm.results. Example:
+ * [ [ sensorLog1, sensorLog2 .. ], [ sensorLog1, sensorLog2, sensorLog3, .. ]]
+ * where each element is a Realm.results<SensorLog>
+ */
+export const extractBreaches = ({ sensorLogs = [], database }) => {
+  if (!(database && sensorLogs.length > 0)) return [];
+
+  let sensorLogStack = [];
+  const breaches = [];
+  const sensorLogsByLocation = {};
+
+  // Sort the sensor logs by timestamp.
+  sensorLogs.sorted('timestamp', true);
+
+  // Group all sensorLogs by location
+  sensorLogs.forEach(sensorLog => {
+    const { location } = sensorLog;
+    // If this sensorLog doesn't have a location,
+    // ignore it
+    if (!location) return;
+    if (!sensorLogsByLocation[location.id]) sensorLogsByLocation[location.id] = [sensorLog];
+    else sensorLogsByLocation[location.id].push(sensorLog);
```

Comment:
change not required, but food for though
```
 sensorLogs.forEach(({ location }) => {  
    if(!location || sensorLogsByLocation[location]) return;
   sensorLogsByLocation[location.id] = sensorLogs.filtered('location.id = $0', location.id).sorted('timestamp')
} 
```

Reply:
done, forgot to do a commit for this, sorry: b8919e997007a723bf88c9af19ab8e12a5807416